internal/venice-common/src/main/java/com/linkedin/venice/annotation/VisibleForTesting.java

```diff
@@ -0,0 +1,18 @@
+package com.linkedin.venice.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+/**
+ * Indicates that a method, class, or field is more visible than otherwise necessary
+ * for the purpose of testing. This is useful for documenting code and communicating
+ * that a specific visibility level (e.g., `public` or `protected`) is intentional
+ * for testing purposes.
+ */
+@Retention(RetentionPolicy.CLASS) // Retained in class files but not available at runtime.
+@Target({ ElementType.METHOD, ElementType.CONSTRUCTOR, ElementType.TYPE, ElementType.FIELD })
+public @interface VisibleForTesting {
```

Comment:
not used in this PR? I tried search the usage but does not find any.

Reply:
Originally this PR had some APIs where it was used. However, I've removed those changes but decided to keep this annotation so that it will be available to use sooner